packages/react/emotion-utils/src/flex.tsx

```diff
@@ -41,19 +41,20 @@ export function flex(
 
 flex.center = (direction?: FlexOptions['direction']) => flex({ justify: 'center', align: 'center', direction });
 
-interface FlexProps<T extends keyof JSX.IntrinsicElements = 'div'> extends AsProps<T>, FlexOptions {}
+export type FlexProps<T extends StringElementType = StringElementType> = AsProps<T> & FlexOptions;
 
-type BaseFlexType = <T extends keyof JSX.IntrinsicElements = 'div'>(
-  props: FlexProps<T> & { ref?: Ref<InferenceHTMLElement<T>> }
-) => ReactElement | null;
+export type BaseFlexType = <T extends StringElementType = StringElementType>(
+  props: FlexProps<T> & Partial<Pick<ComponentPropsWithRef<T>, 'ref'>>
+) => JSX.Element | null;
 
-export const BaseFlex = forwardRef<HTMLElement, FlexProps>(function BaseFlex(props, ref) {
-  const { align = 'stretch', as = 'div', direction = 'row', justify = 'flex-start', ...rest } = props;
-
-  const Component = as as any;
-
-  return <Component ref={ref} css={flex({ align, direction, justify })} {...rest} />;
-}) as BaseFlexType;
+export const BaseFlex: BaseFlexType = forwardRef(function BaseFlex<T extends StringElementType = StringElementType>(
+  props: FlexProps<T>,
+  ref: ComponentPropsWithRef<T>['ref']
+) {
+  const { as = 'div', direction = 'row', justify = 'flex-start', align = 'stretch', ...rest } = props;
+  const As = as;
```

Comment:
I think we can keep the previous name `Component`. What do the name `As` means here?

Reply:
This is a code I left because I was bored haha😁 The name As is a component of 'as prop', and I was wondering if there was a way to shorten the name. But in the end, I changed it to a commonly used name called Component.

Thanks for your eagle eyes!